packages/block-library/src/navigation/editor.scss

```diff
@@ -47,14 +47,27 @@ $navigation-item-height: 46px;
 	}
 
 	&.is-selected,
-	&.has-child-selected {
+	&.has-child-selected,
+	// Show the submenu list when is dragging and drag enter the list item.
+	body.is-dragging-components-draggable &.is-dragging-within {
 		> .wp-block-navigation__container {
 			opacity: 1;
 			visibility: visible;
 		}
 	}
 }
 
+body.is-dragging-components-draggable .wp-block-navigation-link > .wp-block-navigation__container {
```

Comment:
Let's drop the `body` part of the selector if possible, usually just class names are used.